opal-ui/src/components/admin/UsersList.vue

```diff
@@ -0,0 +1,252 @@
+const groupsStore = useGroupsStore();
+const users = computed(() => usersStore.users || []);
+const { t } = useI18n();
+const filter = ref('');
+const loading = ref(false);
+const showAddUser = ref(false);
+const showDelete = ref(false);
+const authenticationType = ref(SubjectCredentialsDto_AuthenticationType.PASSWORD);
+const selectedUser = ref<SubjectCredentialsDto | null>(null);
+
+const columns = [
+  {
+    name: 'name',
+    required: true,
+    label: t('name'),
+    align: 'left',
+    field: 'name',
+    format: (val: string) => val,
+    sortable: true,
+  },
+  {
+    name: 'groups',
+    label: t('groups'),
+    align: 'left',
+    field: 'groups',
+    format: (val: string[]) => (val || []).join(', '),
+    style: 'width: 50%',
+  },
+  {
+    name: 'authentication',
+    label: t('authentication'),
+    align: 'left',
+    field: 'authenticationType',
+    format: (val: string) => t(`auth_types.${val}`),
+  },
+  {
+    name: 'enabled',
+    label: t('enabled'),
+    align: 'center  ',
+    field: 'enabled',
+  },
+];
+
+const initialPagination = ref({
+  sortBy: 'name',
+  descending: false,
+  page: 1,
+  rowsPerPage: 10,
+  minRowsForPagination: 10,
+});
+
+const toolsVisible = ref<{ [key: string]: boolean }>({});
+const filteredUsers = computed(() => filterUsers(users.value, filter.value));
+
+const filterUsers = (rows: any[], terms: string) => {
+  const query = !!filter.value && filter.value.length > 0 ? filter.value.toLowerCase() : '';
+
+  const result = rows.filter((row) => {
+    return Object.values(row).some((val) => {
+      return String(val).toLowerCase().includes(query);
+    });
+  });
+
+  return result;
+};
+
+function onOverRow(row: SubjectCredentialsDto) {
+  toolsVisible.value[row.name] = true;
+}
+
+function onLeaveRow(row: SubjectCredentialsDto) {
+  toolsVisible.value[row.name] = false;
+}
+
+function onEditUser(user: SubjectCredentialsDto) {
+  showAddUser.value = true;
+  selectedUser.value = user;
+  authenticationType.value = user.authenticationType;
+}
+
+async function onDeleteUser(user: SubjectCredentialsDto) {
+  selectedUser.value = user;
+  showDelete.value = true;
+}
+
+async function doDeleteUser() {
+  showDelete.value = false;
+  if (selectedUser.value == null) {
+    return;
+  }
+
+  const toDelete: SubjectCredentialsDto | null = selectedUser.value;
+  selectedUser.value = null;
+
+  Promise.all([usersStore.deleteUser(toDelete), groupsStore.initGroups()]).catch(notifyError);
+}
+
+async function onEnableUser(user: SubjectCredentialsDto) {
+  user.enabled = !user.enabled;
+  usersStore.updateUser(user).catch(notifyError);
+}
+
+function onAddWithPassword() {
+  showAddUser.value = true;
+  authenticationType.value = SubjectCredentialsDto_AuthenticationType.PASSWORD;
+}
+
+function onAddWithCertificate() {
+  showAddUser.value = true;
+  authenticationType.value = SubjectCredentialsDto_AuthenticationType.CERTIFICATE;
+}
+
+async function onUserAdded() {
+  await groupsStore.initGroups();
```

Comment:
why awaiting for initializing following variables?

Reply:
Removed await...